qemu/tests/qemu_guest_agent.py

```diff
@@ -343,6 +343,37 @@ def gagent_verify(self, params, vm):
         self.gagent.verify_responsive()
         logging.info(self.gagent.cmd("guest-info"))
 
+    @error_context.context_aware
+    def gagent_setsebool_value(self, value, params, vm):
+        '''
+        Set selinux boolean 'virt_qemu_ga_read_nonsecurity_files'
+        as 'on' or 'off' for linux guest can access filesystem
+        successfully and restore guest original env when test is over.
+
+        :param value: value of selinux boolean.
+        :param params: Dictionary with the test parameters
+        :param vm: Virtual machine object.
+        '''
+        session = self._get_session(params, vm)
+        self._open_session_list.append(session)
+        if value == 'on':
+            # for accessing filesystem successfully.
+            error_context.context("Set seboolean virt_qemu_ga_read_nonsecurity_files "
+                                  "as on.", logging.info)
+            set_selinux_bool_cmd = params['setsebool_cmd']
+        else:
+            # for restoring linux guest env when test is over.
+            error_context.context("Set seboolean virt_qemu_ga_read_nonsecurity_files "
+                                  "as off.", logging.info)
+            set_selinux_bool_cmd = params['setsebool_back_cmd']
```

Comment:
I understand that you want to clarify that it is environment set up or recovery, but the implementation could be simpler:
cfg:
```
setsebool_cmd = "setsebool virt_qemu_ga_read_nonsecurity_files %s"
```
py:
```
error_context.context("Turn %s seboolean virt_qemu_ga_read_nonsecurity_files" % value, logging.info)
set_selinux_bool_cmd = params['setsebool_cmd'] % value
```